src/components/sessions/session-detail-view/AttendanceTable.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { useEffect, useState } from "react";
+
+import {
+  Button,
+  Checkbox,
+  createMuiTheme,
+  MuiThemeProvider,
+} from "@material-ui/core";
+import CheckIcon from "@material-ui/icons/Check";
+import { makeStyles } from "@material-ui/styles";
+import _ from "lodash";
+import MUIDataTable, {
+  MUIDataTableColumn,
+  MUIDataTableOptions,
+} from "mui-datatables";
+
+import { ClassDetailResponse, ClassRequest } from "api/types";
+import DefaultFieldKey from "constants/DefaultFieldKey";
+
+type AttendanceTableRow = {
+  id: string;
+  first_name: string;
+  last_name: string;
+  [key: string]: string; // array of dates and student's attendance is represented as "yes" or "no"
+};
+
+type AttendanceTableProps = {
+  classObj: ClassDetailResponse;
+  isEditing: boolean;
+  onSubmit: (data: ClassRequest) => void;
+};
+
+const useStyles = makeStyles(() => ({
+  button: {
+    order: -1,
+    height: 36,
+    marginTop: 5,
+    marginRight: 15,
+    textTransform: "none",
+  },
+}));
+
+const monthNames = [
+  "Jan",
+  "Feb",
+  "Mar",
+  "Apr",
+  "May",
+  "Jun",
+  "Jul",
+  "Aug",
+  "Sep",
+  "Oct",
+  "Nov",
+  "Dec",
+];
+
+const AttendanceTable = ({
+  classObj,
+  isEditing,
+  onSubmit,
+}: AttendanceTableProps) => {
+  const classes = useStyles();
+  const [data, setData] = useState<ClassRequest>(_.cloneDeep(classObj));
+  const [tableRows, setTableRows] = useState<AttendanceTableRow[]>([]);
+
+  const handleCheckboxOnClick = (id: number, date: string) => {
+    const dateIndex = data.attendance.findIndex(
+      (element: { date: string }) => element.date === date
+    );
+    const idIndex = data.attendance[dateIndex].attendees.findIndex(
+      (element) => element === id
+    );
+    const newData: ClassDetailResponse = _.cloneDeep(data);
+    if (idIndex === -1) {
+      newData.attendance[dateIndex].attendees.push(id);
+    } else {
+      newData.attendance[dateIndex].attendees.splice(idIndex, 1);
+    }
+    setData(newData);
+  };
+
+  const getMuiTheme = () =>
+    createMuiTheme({
+      overrides: {
+        MUIDataTableToolbar: {
+          actions: {
+            display: "flex",
+            flexDirection: "row",
+            flex: "initial",
+          },
+        },
+      },
+    });
+
+  const options: MUIDataTableOptions = {
+    responsive: "standard",
+    pagination: false,
+    selectableRows: "none",
+    elevation: 0,
+    customToolbar: () => (
+      <Button
+        variant="outlined"
+        color={!isEditing ? "default" : "primary"}
+        onClick={() => onSubmit(data)}
+        className={classes.button}
+      >
+        {!isEditing ? "Take attendance " : "Done attendance "}
+        <CheckIcon />
```

Comment:
this looks a little big compared to the designs - can we apply the following css properties?
```
height: 20px;
width: 20px;
margin-left: 8px;
```